charts/minecraft/templates/datadir-pvc.yaml

```diff
@@ -9,6 +9,11 @@ metadata:
     release: "{{ .Release.Name }}"
     heritage: "{{ .Release.Service }}"
   annotations:
+  {{- if .Values.persistence.storageClass }}
```

Comment:
Throwing out an idea, perhaps users could opt-into the non-annotation based storageClass via a new value condition here.